forge-ai/src/main/java/forge/ai/ability/ChangeZoneAi.java

```diff
@@ -2075,6 +2078,35 @@ public boolean apply(Card card) {
         }
     }
 
+    private boolean doExileSpellLogic(final Player aiPlayer, final SpellAbility sa) {
+        String aiLogic = sa.getParamOrDefault("AILogic", "");
+        SpellAbilityStackInstance top = aiPlayer.getGame().getStack().peek();
+        List<ApiType> dangerousAPI = Arrays.asList(ApiType.DealDamage, ApiType.DamageAll, ApiType.Destroy, ApiType.DestroyAll, ApiType.Sacrifice, ApiType.SacrificeAll);
+        int manaCost = 0;
+        int minCost = 0;
+
+        if (aiLogic.contains(".")) {
+            minCost = Integer.parseInt(aiLogic.substring(aiLogic.indexOf(".") + 1));
+        }
+
+        if (top != null) {
+            SpellAbility topSA = top.getSpellAbility(false);
+            if (topSA != null) {
+                if (topSA.getPayCosts().hasManaCost()) {
+                    manaCost = topSA.getPayCosts().getTotalMana().getCMC();
+                }
+
+                if ((manaCost >= minCost || dangerousAPI.contains(topSA.getApi()))
+                        && topSA.getActivatingPlayer().isOpponentOf(aiPlayer)
+                        && sa.canTargetSpellAbility(topSA)) {
+                    sa.getTargets().add(topSA);
```

Comment:
doesn't it need resetTargets or it will always fail if it can't pay on first try?

Reply:
Yep, good catch! :)